feat(tools): Introduce tool to generate man pages

teodoradriann opened the pull request:
### Prerequisite checklist
  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [x] Updated relevant documentation.

### Description of changes

I updated the auto-generating documentation script located at ```tools/gendocs/main.go``` by adding two new functionalities.

Now, after compiling the program, you can use one of the following flags:
• ```--man or -m:``` Automatically generates all KraftKit documentation as .1 files, located in kraftkit/doc/man.
• ```--install or -i:``` Installs all the files found in the man folder system-wide and updates the mandb.

<img width="633" alt="SCR-20241218-qbxe" src="https://github.com/user-attachments/assets/3badf46d-08b2-4579-9749-6ba7e1dccdc2" />
<img width="353" alt="SCR-20241218-qbzu" src="https://github.com/user-attachments/assets/e5bc719e-1f69-45f0-86cc-12f12edee7e0" />
<img width="1658" alt="SCR-20241218-qccw" src="https://github.com/user-attachments/assets/eab2d53e-ee1d-45aa-9a38-4f2964390ed2" />


GitHub-Closes: #1704

craciunoiuc commented:
Nice, thanks for this @teodoradriann 

Some initial mentions from me:
1. Can you squash your commits and tidy them up? I think 1-2 commits are fine for this use case. I edited your PR title, I guess you can use that one also.
2. Can you move your changes from `gendocs` to a separate `tools/genman/main.go`. The gendocs tool is used for the website to generate things in a certain format. Just copy your things to the separate file and delete everything `gendocs`-related.
3. Add some pictures with the result to the PR description.

That's all from my side, after this is done I'll leave some code comments also

teodoradriann commented:
Okay, will do and update

teodoradriann committed:
feat(tools): Add man pages generator

This is an initial version of the auto-generator of man pages for
kraftkit. This parses help messages and generates man files which
can be afterwards installed.

To generate them, use the new `make` command `make man`.

Signed-off-by: Teodor Adrian Miron <teodormiron75@gmail.com>
Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

craciunoiuc commented:
Hey @teodoradriann, are you up for some more rounds of reviews or do you want me to take over this PR to get it to shape?

Let me know how your availability is looking

Edit: I pushed my fixes (hopefully ok with you)

craciunoiuc committed:
feat(goreleaser): Add man pages to packages for installing

Signed-off-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

nderjung commented on Makefile:
```diff
@@ -238,7 +238,12 @@ properclean: ## Completely clean the repository's build artifacts.
+
+.PHONY: man
+man: OUTDIR ?= $(WORKDIR)/docs/man/
+man: ## Generate Man documentation.
```
`s/Man/manpage/`

nderjung reviewed (approved):
Thanks!

Reviewed-by: Alexander Jung <alex@unikraft.io>
Approved-by: Alexander Jung <alex@unikraft.io>
